src/mango/src/mango_cursor_view.erl

```diff
@@ -107,10 +107,14 @@ execute(#cursor{db = Db, index = Idx} = Cursor0, UserFun, UserAcc) ->
 % check FieldRanges for a, b, c, and d and return
 % the longest prefix of columns found.
 composite_indexes(Indexes, FieldRanges) ->
+    FieldKeys = [Key || {Key, _} <- FieldRanges],
     lists:foldl(fun(Idx, Acc) ->
         Cols = mango_idx:columns(Idx),
         Prefix = composite_prefix(Cols, FieldRanges),
-        [{Idx, Prefix} | Acc]
+        % create a KeysPrefixLength based on how close the prefix of
+        % the index is to the number of keys in the selector
+        KeysPrefixLength = length(FieldKeys) - length(Prefix),
```

Comment:
"NumExtraColumns" seems like it'd be better. Also the comment should be updated to specify that we're looking to use the index with the fewest extra columns. And we should add a comment below in choose_best_index.